apps/api/test/functional/templates.spec.ts

```diff
@@ -0,0 +1,106 @@
+import { faker } from "@faker-js/faker";
+import * as fs from "node:fs";
+import * as path from "node:path";
+
+import { app } from "@src/app";
+import type { Category } from "@src/template/types/template";
+import { env } from "@src/utils/env";
+
+const fakeHeaders = {
+  headers: {
+    "x-ratelimit-remaining": 100
+  }
+};
+
+env.GITHUB_PAT = "ghp_1234567890";
+
+jest.mock("@octokit/rest", () => {
+  return {
+    Octokit: jest.fn().mockImplementation(() => ({
+      rest: {
+        repos: {
+          getBranch: jest.fn().mockImplementation(() => {
+            return Promise.resolve({
+              status: 200,
+              data: {
+                commit: {
+                  sha: faker.git.commitSha()
+                }
+              },
+              ...fakeHeaders
+            });
+          }),
+          getContent: jest.fn().mockImplementation((params: { owner: string; repo: string; path: string }) => {
+            if (params.path === "README.md") {
+              return Promise.resolve({
+                status: 200,
+                data: fs.readFileSync(path.join(__dirname, "..", "mocks/templates", params.owner, params.repo, params.path.toLowerCase()), "utf8"),
+                ...fakeHeaders
+              });
+            }
+
+            return Promise.resolve({
+              status: 200,
+              data: JSON.parse(fs.readFileSync(path.join(__dirname, "..", "mocks/templates", params.owner, params.repo, params.path, "index.json"), "utf8")),
+              ...fakeHeaders
+            });
+          })
+        }
+      }
+    }))
+  };
+});
+
+describe("Templates API", () => {
```

Comment:
thought(non-blocking): in this case testing API doesn't bring much value. The core functionality here is handled by:

```ts
TemplateGalleryService
TemplateFetcherService,
TemplateProcessorService
```

And we should always concentrate our attention on what entity does the most work because then we can apply Pareto principle and cover in details only important things which has higher complexity or higher likelihood of being broken

Reply:
I think we should not do it at least for now because I don't see a usecase. We prebuild templates at docker build time and they don't change often. There is [npm script](https://github.com/akash-network/console/blob/main/apps/api/package.json#L14) which is called by [npm run build](https://github.com/akash-network/console/blob/main/apps/api/package.json#L13).

You can read some of the ideas about testing in our [Testing strategy](https://github.com/akash-network/console/discussions/1130).